subprojects/testing-junit-platform/src/main/java/org/gradle/api/internal/tasks/testing/junitplatform/JUnitPlatformTestClassProcessor.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.tasks.testing.junitplatform;
+
+import org.gradle.api.Action;
+import org.gradle.api.internal.tasks.testing.TestResultProcessor;
+import org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor;
+import org.gradle.api.internal.tasks.testing.junit.TestClassExecutionListener;
+import org.gradle.internal.actor.ActorFactory;
+import org.gradle.internal.id.IdGenerator;
+import org.gradle.internal.time.Clock;
+import org.junit.platform.launcher.Launcher;
+import org.junit.platform.launcher.LauncherDiscoveryRequest;
+import org.junit.platform.launcher.core.LauncherDiscoveryRequestBuilder;
+import org.junit.platform.launcher.core.LauncherFactory;
+
+import static org.junit.platform.engine.discovery.DiscoverySelectors.selectClass;
+
+public class JUnitPlatformTestClassProcessor extends JUnitTestClassProcessor {
+    public JUnitPlatformTestClassProcessor(IdGenerator<?> idGenerator, ActorFactory actorFactory, Clock clock) {
+        super(null, idGenerator, actorFactory, clock);
+    }
+
+    @Override
+    protected Action<String> createTestExecutor(TestResultProcessor threadSafeResultProcessor, TestClassExecutionListener threadSafeTestClassListener) {
+        Launcher launcher = LauncherFactory.create();
+        launcher.registerTestExecutionListeners(new JUnitPlatformTestExecutionListener(threadSafeResultProcessor, clock, idGenerator));
+        return testClassName -> {
```

Comment:
Idea complains about lambda expression in 1.6 language level.

Reply:
Not sure this is the correct solution but works for me: move cursor to the error, Alt+Enter and select "set language level to 1.8".

Reply:
See my comment above: https://github.com/gradle/gradle/pull/4116#discussion_r164232000